pkg/messaging/messaging.go

```diff
@@ -8,22 +8,30 @@ import (
 	"code-intelligence.com/cifuzz/pkg/log"
 )
 
-func ShowServerConnectionMessage(server string) *url.Values {
+type MessagingContext string
 
-	messagesAndParams := []struct {
+const (
+	Run     MessagingContext = "Run"
+	Finding MessagingContext = "Finding"
+)
+
+func ShowServerConnectionMessage(server string, context MessagingContext) *url.Values {
+
+	messagesAndParams := map[string][]struct {
 		message          string
 		additionalParams url.Values
 	}{
-		{
-			message: fmt.Sprintf(`Do you want to persist your findings?
+		"Run": {
```

Comment:
Since you already created the `MessagingContext` type, I would use it here:

```suggestion
	messagesAndParams := map[MessagingContext][]struct {
		message          string
		additionalParams url.Values
	}{
		Run: {
```